xarray/core/datatree.py

```diff
@@ -158,6 +158,19 @@ def _check_alignment(
             _check_alignment(child_path, child_ds, base_ds, child.children)
 
 
+def _check_for_slashes_in_names(variables: Iterable[Hashable]) -> None:
+    offending_variable_names = [
+        name for name in variables if isinstance(name, str) and "/" in name
+    ]
+    if len(offending_variable_names) > 0:
+        raise KeyError(
+            "Given Dataset contains variable names with '/': "
+            f"{offending_variable_names}. "
+            "A Dataset represents a group, and a single group "
+            "cannot have path-like variable names with '/' characters in them. "
+        )
```

Comment:
I think this change addresses @keewis 's criticism in https://github.com/pydata/xarray/pull/9378/files#r1733135830, without implying that `xr.Dataset` also has this restriction, or making references to netCDF / Zarr groups (as not all users of datatree will use those formats).

```suggestion
        raise KeyError(
            "Given variables have names including the '/' character: "
            f"{offending_variable_names}. "
            "Variables stored in DataTree objects cannot have path-like variable names with '/' characters in them."
        )
```